compiler/rustc_target/src/spec/mod.rs

```diff
@@ -1317,6 +1320,120 @@ pub struct Target {
     pub options: TargetOptions,
 }
 
+impl Target {
+    pub fn parse_data_layout<'a>(&'a self) -> Result<TargetDataLayout, TargetDataLayoutErrors<'a>> {
```

Comment:
I don't like that this is in the "spec" section, the syntax is *not* dictated by rustc target specs, it's a LLVM thing.

So maybe we need `TargetDataLayout::parse_from_llvm_datalayout_string` and then this method can keep the extra checks?